adapter/internal/synchronizer/apis_fetcher.go

```diff
@@ -318,7 +318,7 @@ func FetchAPIsFromControlPlane(updatedAPIID string, updatedEnvs []string) {
 				logger.LoggerSync.Debugf("Time Duration for retrying: %v", conf.ControlPlane.RetryInterval*time.Second)
 				time.Sleep(conf.ControlPlane.RetryInterval * time.Second)
 				logger.LoggerSync.Info("Retrying to fetch API data from control plane.")
-				FetchAPIs(&updatedAPIID, finalEnvs, c)
+				go FetchAPIs(&updatedAPIID, finalEnvs, c)
```

Comment:
why do we need a go-routine here?

Reply:
Idea was to use a goroutine to make sure a single point of failure don't end up crashing the adapter. But since this call is already under a goroutine, we could remove it. @Rajith90 based on this logic, can we remove this?

Reply:
We need a one go routine at the root level, otherwise we can't use the panic recover  function without a go routine AFAIU

Reply:
Verified by replicating a panic scenario. We can still recover successfully even after removing this goroutine. Hence removed this as well. Refer [218fe21](https://github.com/wso2/product-microgateway/pull/2103/commits/218fe21463db20eee3387a5310646aaa45a7d860)